### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by zacblazic**
This is the initial set of commits that aims to adjust the default signal handling of nginx to ensure it performs a graceful shutdown when `docker stop` is invoked.

**Background**

Depending on the [signal used to stop nginx](http://nginx.org/en/docs/control.html), its behaviour will be different:

Signal | Action
-- | --
`TERM`, `INT` | Fast shutdown
`QUIT` | Graceful shutdown
`HUP` | Changing configuration, keeping up with a changed time zone (only for FreeBSD and Linux), starting new worker processes with a new configuration, graceful shutdown of old worker processes
`USR1` | Re-opening log files
`USR2` | Upgrading an executable file
`WINCH` | Graceful shutdown of worker processes

Currently, when kubernetes terminates a pod, it is translated to issuing a [`docker stop`](https://docs.docker.com/engine/reference/commandline/stop/) against the underlying node. The default behaviour of [`docker stop`](https://docs.docker.com/engine/reference/commandline/stop/) is to send a `SIGTERM` to the main process in the container.

This means that by default, a pod / container running nginx will receive a `SIGTERM` when rolling out a new deployment or scaling down and thus will perform a "fast" shutdown, and not a graceful shutdown.

**Solution**

What we really want is for `docker stop` to instead send a `SIGQUIT` to the nginx container. Implementing this is pretty straight-forward thanks to the [`STOPSIGNAL`](https://docs.docker.com/engine/reference/builder/#stopsignal) feature of Docker which was used here:

```dockerfile
FROM nginx:1.16.1-alpine

STOPSIGNAL SIGQUIT
```

**Caveats**

In https://github.com/nginxinc/docker-nginx/issues/46, the default stop signal was changed to `SIGQUIT` however it was later reverted after some [backlash](https://github.com/nginxinc/docker-nginx/issues/46#issuecomment-299486357). While there is some merit to the revert regarding the use of unix sockets, our use case does not involve unix sockets and therefore having a graceful shutdown is preferred.

### CODE REVIEW

File: config/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    default_type  application/octet-stream;
+
+    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                      '$status $body_bytes_sent "$http_referer" '
+                      '"$http_user_agent" "$http_x_forwarded_for"';
+
+    access_log  /var/log/nginx/access.log  main;
+
+    sendfile        on;
+    #tcp_nopush     on;
```
Comment (hadrianvalentine) [inline config/nginx.conf]: perhaps we can remove the commented config?

File: config/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    keepalive_timeout  65;
+
+    #gzip  on;
```
Comment (hadrianvalentine) [inline config/nginx.conf]: Same as https://github.com/Intellection/docker-nginx/pull/1/files#r330548357

File: config/nginx.conf (hunk 1 above)
Comment (zacblazic) [reply config/nginx.conf]: Resolved in https://github.com/Intellection/docker-nginx/pull/1/commits/88290f8275c330515d657e1226a4c179422f7dde.

File: config/nginx.conf (hunk 2 above)
Comment (zacblazic) [reply config/nginx.conf]: Resolved in https://github.com/Intellection/docker-nginx/pull/1/commits/88290f8275c330515d657e1226a4c179422f7dde.

**Review by hadrianvalentine — APPROVED** [review]

**Pull request merged**